lean/elf.lean

```diff
@@ -322,29 +325,85 @@ end word
 ------------------------------------------------------------------------
 -- phdr
 
-structure phdr_type :=
-(val : uint32)
+inductive phdr_type : Type
+  | PT_NULL    : phdr_type
+  | PT_LOAD    : phdr_type
+  | PT_DYNAMIC : phdr_type
+  | PT_INTERP  : phdr_type
+  | PT_NOTE    : phdr_type 
+  | PT_SHLIB   : phdr_type
+  | PT_PHDR    : phdr_type
+  | PT_PROC    : uint32 -> phdr_type
```

Comment:
I'm inclined to keep the original definition for this, and flags, but add definitions for constants e.g.,

  `def PT_NULL : phdr_type := \{0\}`
 
We might then use an rb_map to match phdr types to the string.

Reply:
I can revert the change, but am interested to know why you prefer a uint32 to the sum type?

Reply:
The `uint32` is compact and isomorphic to the value stored on disk.  It also treats the OS/processor specific values uniformly as predefined ones.